From 1bd5c463377b0e54dc3bed840906a8df37bf6f7e Mon Sep 17 00:00:00 2001
From: Mike Fleetwood <mike.fleetwood@googlemail.com>
Date: Sat, 1 Oct 2016 16:40:23 +0100
Subject: [PATCH 52/53] tests: t3310-flags.sh: Add tests for remaining table
 types

Add test of flags for remaining table types: aix, amiga, pc98, sun and
loop.  Note that support of writing AIX tables is not yet implemented in
parted and LOOP tables don't support partitions nor flags.

Signed-off-by: Brian C. Lane <bcl@redhat.com>
---
 tests/t3310-flags.sh | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/tests/t3310-flags.sh b/tests/t3310-flags.sh
index e449589..e97c3b9 100644
--- a/tests/t3310-flags.sh
+++ b/tests/t3310-flags.sh
@@ -25,10 +25,16 @@ extract_flags()
   perl -nle '/^[^:]*:4096s:6143s:2048s::[^:]*:(.+);$/ and print $1' "$@"
 }
 
-for table_type in bsd dvh gpt mac msdos; do
+for table_type in aix amiga bsd dvh gpt mac msdos pc98 sun loop; do
   ptn_num=1
 
   case $table_type in
+    aix)   # Support for writing AIX disk labels and adding partitions
+           # is not yet implemented.
+           continue
+           ;;
+    amiga) primary_or_name='PTNNAME'
+           ;;
     bsd)   primary_or_name=''
            ;;
     dvh)   primary_or_name='primary'
@@ -42,6 +48,14 @@ for table_type in bsd dvh gpt mac msdos; do
            ;;
     msdos) primary_or_name='primary'
            ;;
+    pc98)  primary_or_name='PTNNAME'
+           ;;
+    sun)   primary_or_name=''
+           ;;
+    loop)  # LOOP table doesn't support creation of a partition nor any
+           # flags.
+           continue
+           ;;
   esac
 
   n_sectors=8192
-- 
2.7.4

